plugin-server/src/cdp/hog-exception-executor.ts

```diff
@@ -0,0 +1,476 @@
+    //         try {
+    //             if (hogFunction.filters?.bytecode) {
+    //                 const filterResult = exec(hogFunction.filters.bytecode, {
+    //                     globals: filtersGlobals,
+    //                     timeout: DEFAULT_TIMEOUT_MS,
+    //                     maxAsyncSteps: 0,
+    //                 })
+
+    //                 if (typeof filterResult.result === 'boolean' && filterResult.result) {
+    //                     matchingFunctions.push(hogFunction)
+    //                     return
+    //                 }
+    //             }
+    //         } catch (error) {
+    //             // TODO: This should be reported as a log or metric
+    //             status.error('🦔', `[HogExecutor] Error filtering function`, {
+    //                 hogFunctionId: hogFunction.id,
+    //                 hogFunctionName: hogFunction.name,
+    //                 error: error.message,
+    //             })
+    //         }
+
+    //         nonMatchingFunctions.push(hogFunction)
+    //     })
+
+    //     status.debug(
+    //         '🦔',
+    //         `[HogExecutor] Found ${Object.keys(matchingFunctions).length} matching functions out of ${
+    //             Object.keys(allFunctionsForTeam).length
+    //         } for team`
+    //     )
+
+    //     return {
+    //         nonMatchingFunctions,
+    //         matchingFunctions,
+    //     }
+    // }
+
+    /**
+     * Intended to be invoked as a starting point from an event
+     */
+    async executeFunction(
+        event: HogFunctionInvocationGlobals,
+        functionOrId: HogFunctionType | HogFunctionType['id']
+    ): Promise<HogFunctionInvocationResult | undefined> {
+        const hogFunction =
+            typeof functionOrId === 'string'
+                ? this.hogFunctionManager.getTeamHogFunction(event.project.id, functionOrId)
+                : functionOrId
+
+        if (!hogFunction) {
+            return
+        }
+
+        const eventFingerprint = event.event.properties.$exception_fingerprint
+
+        // TODO: would this be better as a HogQL fetch request?
+        const items = (
+            await this.postgres.query<HogFunctionType>(
+                PostgresUse.COMMON_READ,
+                `
+            SELECT fingerprint
+            FROM posthog_errortrackinggroup
+            WHERE (fingerprint = '{$1}'::text[]) OR (merged_fingerprints @> '{$1}'::text[])
+        `,
+                [eventFingerprint.join(',')],
+                'fetchErrorTrackingGroup'
+            )
+        ).rows
+
+        const fingerprint = items.length > 0 ? items[0] : eventFingerprint
+
+        // Add the source of the trigger to the globals
+        const modifiedGlobals: HogFunctionInvocationGlobals = {
+            ...event,
+            event: {
+                ...event.event,
+                properties: {
+                    ...event.event.properties,
+                    $exception_fingerprint: fingerprint,
```

Comment:
Should we be changing the property or adding a new global variable to the state?